script/SafeTxDataBuilder.sol

```diff
@@ -25,40 +28,33 @@ contract SafeTxDataBuilder is Script, SignatureDecoder {
     //     "SafeTx(address to,uint256 value,bytes data,uint8 operation,uint256 safeTxGas,uint256 baseGas,uint256 gasPrice,address gasToken,address refundReceiver,uint256 nonce)"
     // );
     bytes32 private constant SAFE_TX_TYPEHASH = 0xbb8310d486368db6bd6f849402fdd73ad53d316b5a4b2644ad6efe0f941286d8;
+    bytes NEWLINE_CHAR = bytes("\n");
 
     string ROOT = vm.projectRoot();
     string SIGNATURES_DIR = string.concat(ROOT, "/signatures/");
+
+    string TX_FILE = string.concat(SIGNATURES_DIR, "tx.json");
+    string HASH_DATA_FILE = string.concat(SIGNATURES_DIR, "hashData.txt");
     string SIGNATURES_FILE = string.concat(SIGNATURES_DIR, "signatures.txt");
-    string DATA_FILE = string.concat(SIGNATURES_DIR, "data.txt");
-    bytes NEWLINE_CHAR = bytes("\n");
 
     GnosisSafe immutable SAFE;
 
     constructor(address payable safe) {
         SAFE = GnosisSafe(safe);
     }
 
-    function loadData() internal view returns (bytes memory) {
-        bytes memory rawData = bytes(vm.readFile(DATA_FILE));
-        uint256 rawDataLength = rawData[rawData.length - 1] == NEWLINE_CHAR[0] ? rawData.length - 1 : rawData.length;
-
-        assembly {
-            mstore(rawData, rawDataLength)
-        }
-
-        return vm.parseBytes(string(rawData));
-    }
-
     function loadSafeTxData() internal returns (SafeTxData memory txData) {
-        txData.to = vm.envAddress("TO");
-        txData.value = vm.envOr("VALUE", uint256(0));
-        txData.data = loadData();
-        txData.operation = Enum.Operation(vm.envOr("OPERATION", uint256(0)));
-        txData.safeTxGas = 0;
-        txData.baseGas = 0;
-        txData.gasPrice = 0;
-        txData.gasToken = address(0); // ETH
-        txData.refundReceiver = payable(address(0)); // tx.origin
+        string memory json = vm.readFile(TX_FILE);
+
+        txData.to = json.readAddress("$.to");
+        txData.value = json.readUint("$.value");
+        txData.data = json.readBytes("$.data");
+        txData.operation = Enum.Operation(json.readUint("$.operation"));
```

Comment:
accepting "call"/"delegatecall" would be great

Reply:
It's already the case ;) I'll document it